src/django_components/component_media.py

```diff
@@ -0,0 +1,375 @@
+                    media.css[media_type] = [path_list]  # type: ignore
+
+            # Validate CSS media types
+            # See https://developer.mozilla.org/en-US/docs/Web/CSS/@media#media_types
+            # And https://docs.djangoproject.com/en/5.0/topics/forms/media/#css
+            for media_type in media.css:
+                if media_type not in CSS_MEDIA_TYPES:
+                    raise ValueError(
+                        "Keys of Media.css must be valid CSS media types "
+                        f"({', '.join(CSS_MEDIA_TYPES)}), got {media_type}. "
+                    )
+
+    if hasattr(media, "js") and media.js:
+        # Allow: class Media: js = "script.js"
+        if _is_media_filepath(media.js):
+            media.js = [media.js]  # type: ignore[list-item]
+
+
+def _map_media_filepaths(media: ComponentMediaInput, map_fn: Callable[[Any], Any]) -> None:
+    if hasattr(media, "css") and media.css:
+        if not isinstance(media.css, dict):
+            raise ValueError(f"Media.css must be a dict, got {type(media.css)}")
+
+        for media_type, path_list in media.css.items():
+            media.css[media_type] = list(map(map_fn, path_list))  # type: ignore[assignment]
+
+    if hasattr(media, "js") and media.js:
+        if not isinstance(media.js, (list, tuple)):
+            raise ValueError(f"Media.css must be a list, got {type(media.css)}")
+
+        media.js = list(map(map_fn, media.js))
+
+
+def _is_media_filepath(filepath: Any) -> bool:
+    if callable(filepath):
+        return True
+
+    if isinstance(filepath, SafeData) or hasattr(filepath, "__html__"):
+        return True
+
+    elif isinstance(filepath, (Path, os.PathLike)) or hasattr(filepath, "__fspath__"):
+        return True
+
+    if isinstance(filepath, bytes):
+        return True
+
+    if isinstance(filepath, str):
+        return True
+
+    return False
+
+
+def _normalize_media_filepath(filepath: Any) -> Union[str, SafeData]:
+    if callable(filepath):
+        filepath = filepath()
+
+    if isinstance(filepath, SafeData) or hasattr(filepath, "__html__"):
+        return filepath
+
+    if isinstance(filepath, (Path, os.PathLike)) or hasattr(filepath, "__fspath__"):
+        filepath = filepath.__fspath__()
+
+    if isinstance(filepath, bytes):
+        filepath = filepath.decode("utf-8")
+
+    if isinstance(filepath, str):
+        return filepath
+
+    raise ValueError(
+        "Unknown filepath. Must be str, bytes, PathLike, SafeString, or a function that returns one of the former"
+    )
+
+
+def _resolve_component_relative_files(attrs: MutableMapping) -> None:
+    """
+    Check if component's HTML, JS and CSS files refer to files in the same directory
+    as the component class. If so, modify the attributes so the class Django's rendering
+    will pick up these files correctly.
+    """
+    # First check if we even need to resolve anything. If the class doesn't define any
+    # JS/CSS files, just skip.
```

Comment:
`_resolve_component_relative_files` and `_get_dir_path_from_component_path` didn't change, just added the check here to skip the relative path resolution logic when the component class doesn't define any files.